src/allmydata/test/test_inotify.py

```diff
@@ -0,0 +1,534 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Tests for the inotify-alike implementation L{allmydata.watchdog}.
+"""
+
+# Note: See https://twistedmatrix.com/trac/ticket/8915 for a proposal
+# to avoid all of this duplicated code from Twisted.
+
+import sys
+
+from twisted.internet import defer, reactor
+from twisted.python import filepath, runtime
+from twisted.trial import unittest
+
+from allmydata.frontends.magic_folder import get_inotify_module
+
+inotify = get_inotify_module()
+
+
+class INotifyTests(unittest.TestCase):
+    """
+    Define all the tests for the basic functionality exposed by
+    L{inotify.INotify}.
+    """
+    def skip_if_windows(self):
+        if runtime.platformType == "win32":
+            raise unittest.SkipTest("twistd does not fork under windows")
+
+    def setUp(self):
+        self.ignore_count = 0
+        self.dirname = filepath.FilePath(self.mktemp())
+        self.dirname.createDirectory()
+        self.inotify = inotify.INotify()
+        self.inotify.startReading()
+        self.addCleanup(self.inotify.stopReading)
+        self.skip_if_windows()
```

Comment:
Move this up to the beginning of the `setUp` method?

Or do the skip differently:

```
class INotifyTests(unittest.TestCase):
    if runtime.platformType == "win32":
        skip = "twistd does not fork under windows"

    ...
```

I generally prefer setting the skip attribute to raising `SkipTest` during test execution - if it is possible to statically determine the skip that seems to be superior.